src/item/enchantment/EnchantmentHelper.php

```diff
@@ -0,0 +1,219 @@
+		// Random bonus for enchanting power between 0.85 and 1.15
+		$bonus = 1 + ($random->nextFloat() + $random->nextFloat() - 1) * 0.15;
+		$enchantingPower = (int) round($enchantingPower * $bonus);
+
+		$resultEnchantments = [];
+		$availableEnchantments = self::getAvailableEnchantments($enchantingPower, $inputItem);
+
+		$lastEnchantment = self::getRandomWeightedEnchantment($random, $availableEnchantments);
+		if($lastEnchantment !== null){
+			$resultEnchantments[] = $lastEnchantment;
+
+			// With probability (power + 1) / 50, continue adding enchantments
+			while($random->nextFloat() <= ($enchantingPower + 1) / 50){
+				// Remove from the list of available enchantments anything that conflicts
+				// with previously-chosen enchantments
+				$availableEnchantments = array_filter(
+					$availableEnchantments,
+					function(EnchantmentInstance $e) use ($lastEnchantment){
+						return $e->getType() !== $lastEnchantment->getType() &&
+							$e->getType()->isCompatibleWith($lastEnchantment->getType());
+					}
+				);
+
+				$lastEnchantment = self::getRandomWeightedEnchantment($random, $availableEnchantments);
+				if($lastEnchantment === null){
+					break;
+				}
+
+				$resultEnchantments[] = $lastEnchantment;
+				$enchantingPower >>= 1;
+			}
+		}
+
+		return new EnchantmentOption($requiredXpLevel, self::getRandomOptionName($random), $resultEnchantments);
+	}
+
+	/**
+	 * @return EnchantmentInstance[]
+	 */
+	private static function getAvailableEnchantments(int $enchantingPower, Item $item) : array{
+		$list = [];
+
+		foreach(EnchantingTableOptionRegistry::getInstance()->getAvailableEnchantments($item) as $enchantment){
+			for($lvl = $enchantment->getMaxLevel(); $lvl > 0; $lvl--){
+				if($enchantingPower >= $enchantment->getMinEnchantingPower($lvl) &&
+					$enchantingPower <= $enchantment->getMaxEnchantingPower($lvl)
+				){
+					$list[] = new EnchantmentInstance($enchantment, $lvl);
+					break;
+				}
+			}
+		}
+
+		return $list;
+	}
+
+	/**
+	 * @param EnchantmentInstance[] $enchantments
+	 */
+	private static function getRandomWeightedEnchantment(Random $random, array $enchantments) : ?EnchantmentInstance{
+		if(count($enchantments) === 0){
+			return null;
+		}
+
+		$totalWeight = 0;
+		foreach($enchantments as $enchantment){
+			$totalWeight += $enchantment->getType()->getRarity();
+		}
+
+		$result = null;
+		$randomWeight = $random->nextRange(1, $totalWeight);
+
+		foreach($enchantments as $enchantment){
+			$randomWeight -= $enchantment->getType()->getRarity();
+
+			if($randomWeight <= 0){
+				$result = $enchantment;
+				break;
+			}
+		}
+
+		return $result;
+	}
+
+	private static function getRandomOptionName(Random $random) : string{
+		$symbols = range("a", "z");
+		$name = "";
+
+		for($i = $random->nextRange(5, 15); $i > 0; $i--){
+			$name .= $symbols[$random->nextBoundedInt(count($symbols))];
```

Comment:
This seems like a weird way to generate random gibberish. A simpler way would be `chr($random->nextRange(ord("a"), ord("z")))`